backend/tests/test_parser.py

```diff
@@ -1,68 +1,151 @@
+            "is_complete": True,
+            "resolution": ["1080p"],
+            "quality": ["DVD-Rip"],
+            "season": list(range(1, 29)),
+        },
+    ),
+]
+
+test_ids = ["FullQualityCheck", "SeasonRangeCheck"]
+
+CUSTOM_RANKS = sm.settings.ranking.custom_ranks
+
+
+def test_valid_torrent_from_item():
+    ranking_model = DefaultRanking()
+    torrent = Torrent(
+        ranking_model=ranking_model,
+        raw_title="The Walking Dead S05E03 720p HDTV x264-ASAP[ettv]",
+        infohash="1234567890",
     )
-    assert parsed_data["is_dual_audio"] is True
-
-
-def test_parse_dual_audio_absent(parser):
-    parsed_data = parser.parse(item=None, string="Series S01E02 720p BluRay x264")
-    assert parsed_data["is_dual_audio"] is False
-
-
-def test_parse_complete_series_detected(parser):
-    parsed_data = parser.parse(item=None, string="The Complete Series Box Set 1080p")
-    assert parsed_data["is_complete"] is True
-
-
-def test_parse_complete_series_not_detected(parser):
-    parsed_data = parser.parse(item=None, string="Single.Movie.2020.1080p.BluRay")
-    assert parsed_data["is_complete"] is False
-
-
-def test_parse_unwanted_quality_detected(parser):
-    parsed_data = parser.parse(item=None, string="Low.Quality.Movie.CAM.2020")
-    assert parsed_data["is_unwanted_quality"] is True
-
 
-def test_parse_unwanted_quality_not_detected(parser):
-    parsed_data = parser.parse(item=None, string="High.Quality.Movie.1080p.2020")
-    assert parsed_data["is_unwanted_quality"] is False
+    assert isinstance(torrent, Torrent)
+    assert isinstance(torrent.parsed_data, ParsedMediaItem)
+    assert torrent.raw_title == "The Walking Dead S05E03 720p HDTV x264-ASAP[ettv]"
+    assert torrent.infohash == "1234567890"
+    assert torrent.parsed_data.parsed_title == "The Walking Dead"
+    assert torrent.parsed_data.fetch is True
+    assert torrent.rank == 163, f"Rank was {torrent.rank} instead of 163"
+
+
+def test_check_title_match():
+    """Test the check_title_match function"""
+    from program.versions.parser import check_title_match
+
+    assert check_title_match("Damsel", "Damsel") is True
+    assert check_title_match("American Horror Story", "American Story Horror") is False
+
+
+@pytest.mark.parametrize("raw_title, expected", test_data, ids=test_ids)
+def test_parsed_media_item_properties(raw_title: str, expected: dict):
+    item = ParsedMediaItem(raw_title=raw_title)
+    for key, value in expected.items():
+        assert (
+            getattr(item, key) == value
+        ), f"Attribute {key} failed for raw_title: {raw_title}"
+
+
+def test_episode_parsing():
+    test_cases = [
+        ("The Simpsons S01E01 1080p BluRay x265 HEVC 10bit AAC 5.1 Tigole", [1]),
+        ("The Simpsons S01E01E02 1080p BluRay x265 HEVC 10bit AAC 5.1 Tigole", [1, 2]),
+        ("The Simpsons S01E01-E02 1080p BluRay x265 HEVC 10bit AAC 5.1 Tigole", [1, 2]),
+        # Looks like it doesn't parse past the first 2 episodes
```

Comment:
If this is the case then parse_episodes is broken, please fix it.

Reply:
That did exactly what it was doing before lol